src/lib/database/entries.ts

```diff
@@ -53,7 +56,18 @@ export async function fetchEntries(options: FetchEntriesOptions): Promise<Entry[
 			].filter(Boolean) as Visibility[],
 		},
 	}
+
 	if (searchQuery) searchFilter.$text = { $search: searchQuery }
+	if (tags) {
+		if (tags.length === 1) {
+			searchFilter.tags = tags[0]
+		} else {
+			//may result in unexpected behavior (I think this looks for an exact match, in order?)
+			searchFilter.tags = {
+				$all: tags,
+			}
+		}
```

Comment:
$all checks if all of the tags from the query are also in the document. It doesn't care if the document has extra tags and if they're not in order, so it's preferred to searching without $all (which *does* check for an exact match)
```suggestion
		searchFilter.tags = {
			$all: tags,
		}
```